dagger/pipeline/tasks/batch_task.py

```diff
@@ -63,7 +65,10 @@ def __init__(self, name, pipeline_name, pipeline, job_config):
 
         self._executable = self.parse_attribute("executable")
         self._executable_prefix = self.parse_attribute("executable_prefix") or ""
-        job_name = "{}-{}".format(pipeline.name, self.parse_attribute("job_name"))
+        job_name_raw = f"""{pipeline.name.replace("-", "/")}/{self.parse_attribute("job_name")}"""
```

Comment:
pipeline has a directory attribute as well, so you don't need to convert back from pipeline.name but you can use pipeline.directory instead.